### PR TITLE
GH-619 update user documentation.

### PR DESCRIPTION
**Pull request by danielmkn**

### CODE REVIEW

File: docs/resources/managed_user.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ subcategory: "User"
 Provides an Artifactory managed user resource. This can be used to create and maintain Artifactory users. For example, service account where password is known and managed externally.
 
 Unlike `artifactory_unmanaged_user` and `artifactory_user`, the `password` attribute is required and cannot be empty.
+Consider using a separate provider to generate and manage passwords. 
+
+~> The password is stored in the Terraform state file. Make sure you secure it.
```
Comment (alexhung) [inline docs/resources/managed_user.md:11]: Let's link to this https://developer.hashicorp.com/terraform/language/state/sensitive-data

File: docs/resources/unmanaged_user.md [hunk 2]
```diff
@@ -30,7 +34,7 @@ The following arguments are supported:
 * `profile_updatable` - (Optional) When set, this user can update his profile details (except for the password. Only an administrator can update the password). Default value is `true`.
 * `disable_ui_access` - (Optional) When set, this user can only access Artifactory through the REST API. This option cannot be set if the user has Admin privileges. Default value is `true`.
 * `internal_password_disabled` - (Optional) When set, disables the fallback of using an internal password when external authentication (such as LDAP) is enabled.
-* `groups` - (Optional) List of groups this user is a part of.
+* `groups` - (Optional) List of groups this user is a part of. **Notes:** If this attribute is not specified then user's group membership set to empty. User will not be part of default "readers" group automatically.
```
Comment (alexhung) [inline docs/resources/unmanaged_user.md:37]: 👍🏼

File: docs/resources/unmanaged_user.md [hunk 3]
```diff
@@ -3,10 +3,14 @@ subcategory: "User"
...
-Provides an Artifactory unmanaged user resource. This can be used to create and maintain Artifactory users.
-
+Provides an Artifactory unmanaged user resource. This can be used to create and manage Artifactory users.
+The password is required field by the [Artifactory API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceUser), but we made it optional in this resource to accommodate the scenario, when the password is not needed and will be reset by the actual user later.  
 When the optional attribute `password` is omitted, a random password is generated according to current Artifactory password policy.
 
+~> The generated password won't be stored in the TF state and can not be recovered. The user must reset the password to be able to log in. An admin can always generate the access key for the user as well. The password change won't trigger the state drift.
+
+~> This resource is an alias for `artifactory_user` [resource](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/user), it is identical and was added for clarity and compatibility purpose. 
+
```
Comment (alexhung) [inline docs/resources/unmanaged_user.md:13]: Let's add a short sentence/paragraph to recommend *not* to use this resource unless their use cases fit this resource. And recommend our users to use `artifactory_managed_user` instead. Same for the original `artifactory_user` resource.

**Review by alexhung — CHANGES_REQUESTED** [review]
Let's add more text to discourage the use of both `artifactory_user` and `artifactory_unmanaged_user` resources.

File: docs/resources/unmanaged_user.md [hunk 4]
```diff
@@ -3,10 +3,14 @@ subcategory: "User"
 ---
 # Artifactory Unmanaged User Resource
 
-Provides an Artifactory unmanaged user resource. This can be used to create and maintain Artifactory users.
-
+Provides an Artifactory unmanaged user resource. This can be used to create and manage Artifactory users.
```
Comment (Turhsus) [inline docs/resources/unmanaged_user.md:6]: So I know why its called unmanaged, but it feels confusing to me to call something an `unmanaged` resource and in the next sentence say `create and manage` 

Thoughts?

**Review by alexhung — APPROVED** [review]

**Comment by alexhung** [comment]
@danielmkn We should also consider giving example of the use case for `artifactory_unmanaged_user` from the discussion thread.

**Review by mritunjaykumar — DISMISSED** [review]
LGTM

File: docs/resources/unmanaged_user.md [hunk 5]
```diff
@@ -3,10 +3,14 @@ subcategory: "User"
...
+The password is required field by the [Artifactory API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceUser), but we made it optional in this resource to accommodate the scenario, when the password is not needed and will be reset by the actual user later.  
 When the optional attribute `password` is omitted, a random password is generated according to current Artifactory password policy.
 
+~> The generated password won't be stored in the TF state and can not be recovered. The user must reset the password to be able to log in. An admin can always generate the access key for the user as well. The password change won't trigger the state drift.
```
Comment (Turhsus) [inline docs/resources/unmanaged_user.md:10]: nit: `won't **trigger state** drift`

File: docs/resources/unmanaged_user.md [hunk 6]
```diff
@@ -3,10 +3,14 @@ subcategory: "User"
...
+
+~> This resource is an alias for `artifactory_user` [resource](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/user), it is identical and was added for clarity and compatibility purpose. We don't recommend to use this resource unless there is a specific use case for it. Recommended resource is `artifactory_managed_user`.
```
Comment (Turhsus) [inline docs/resources/unmanaged_user.md:12]: nit: `compatibility purpose**s** `

File: docs/resources/unmanaged_user.md (hunk 2 above)
Comment (Turhsus) [inline docs/resources/unmanaged_user.md:37]: nit: `then **the** user's group membership **is** set`

File: docs/resources/user.md [hunk 7]
```diff
@@ -4,8 +4,10 @@ subcategory: "User"
 # Artifactory User Resource
 
 Provides an Artifactory user resource. This can be used to create and manage Artifactory users.
+The password is required field by the [Artifactory API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceUser), but we made it optional in this resource to accommodate the scenario, when the password is not needed and will be reset by the actual user later.  
+When the optional attribute `password` is omitted, a random password is generated according to current Artifactory password policy. 
 
-When the optional attribute `password` is omitted, a random password is generated according to current Artifactory password policy.
+~> The generated password won't be stored in the TF state and can not be recovered. The user must reset the password to be able to log in. An admin can always generate the access key for the user as well. The password change won't trigger the state drift. We don't recommend to use this resource unless there is a specific use case for it. Recommended resource is `artifactory_managed_user`.
```
Comment (Turhsus) [inline docs/resources/user.md:10]: Same changes as above

File: pkg/artifactory/resource/user/resource_artifactory_user.go [hunk 8]
```diff
@@ -33,7 +33,7 @@ func ResourceArtifactoryUser() *schema.Resource {
 
 		Schema: userSchema,
 
-		Description: "Provides an Artifactory unmanaged user resource. This can be used to create and manage Artifactory users. Password is optional and one will be automatically generated.",
+		Description: "Provides an Artifactory user resource. This can be used to create and manage Artifactory users. The password is required field by the [Artifactory API](https://www.jfrog.com/confluence/display/JFROG/Artifactory+REST+API#ArtifactoryRESTAPI-CreateorReplaceUser), but we made it optional in this resource to accommodate the scenario, when the password is not needed and will be reset by the actual user later. When the optional attribute `password` is omitted, a random password is generated according to current Artifactory password policy.",
```
Comment (Turhsus) [inline pkg/artifactory/resource/user/resource_artifactory_user.go:36]: nit: `The password is **a** required field`

**Review by Turhsus — CHANGES_REQUESTED** [review]
Left some comments requesting grammatical changes. 

Things in between double asterisks like **this** are what could/should be changed.

**Review by MahithaB — DISMISSED** [review]

File: docs/resources/unmanaged_user.md (hunk 4 above)
Comment (danielmkn) [reply docs/resources/unmanaged_user.md:6]: I think it's fine.

**Review by Turhsus — APPROVED** [review]
LGTM

**Review by alexhung — APPROVED** [review]

**Pull request merged**